core/iwasm/interpreter/wasm_interp.h

```diff
@@ -26,6 +26,8 @@ typedef struct WASMInterpFrame {
     /* Instruction pointer of the bytecode array.  */
     uint8 *ip;
 
+    uint32 func_index;
```

Comment:
why do we need it? we already have function above.

Reply:
so now it's calculated like this https://github.com/bytecodealliance/wasm-micro-runtime/blob/b6dea221a609606ce6c3dcaa0f4d0399ea5bc4dc/core/iwasm/interpreter/wasm_interp_fast.c#L1210
which requires dereferencing `module_inst->e->functions`. Always or almost always those pointers remain the same.

But if `wasm_table_get_func_inst` is called then functions pointer gets updated as I can see in https://github.com/bytecodealliance/wasm-micro-runtime/blame/b6dea221a609606ce6c3dcaa0f4d0399ea5bc4dc/core/iwasm/aot/aot_runtime.c#L1369

Reply:
if it's always the case not clear why [this line](https://github.com/bytecodealliance/wasm-micro-runtime/blame/b6dea221a609606ce6c3dcaa0f4d0399ea5bc4dc/core/iwasm/aot/aot_runtime.c#L1369) exists then

Reply:
ok, it seems if this code `module->import_function_count + module->function_count` can never be 0 then yeah it's never modified

But can there be a wasm module with this sum equal 0?

Reply:
I reread the logic there
- [the pointer modified](https://github.com/bytecodealliance/wasm-micro-runtime/blame/b6dea221a609606ce6c3dcaa0f4d0399ea5bc4dc/core/iwasm/aot/aot_runtime.c#L1369) only if pointer is not allocated and `module->import_func_count + module->func_count` equals 0.
- but [the pointer is not allocated](https://github.com/bytecodealliance/wasm-micro-runtime/blob/b6dea221a609606ce6c3dcaa0f4d0399ea5bc4dc/core/iwasm/interpreter/wasm_runtime.c#L2555) only if `module->import_func_count + module->func_count` equals 0.

=> this branch of code with allocating `functions` inside of `wasm_table_get_func_inst` is never executed in fact

Reply:
I removed thios extra field and added an explanation comment, why it's safe to calculate manually